CMakeLists.txt

```diff
@@ -23,7 +23,7 @@ project(Marl C CXX ASM)
 ###########################################################
 option(MARL_WARNINGS_AS_ERRORS "Treat warnings as errors" OFF)
 option(MARL_BUILD_EXAMPLES "Build example applications" OFF)
-option(MARL_BUILD_TESTS "Build tests" ON)
+option(MARL_BUILD_TESTS "Build tests" OFF)
```

Comment:
Any particular reason to have tests off by default?
I am used to other projects that will have tests on by default if you have gtest, and off if you do not.

Reply:
I tried to explain my reasoning in the CL description:

> Marl is a library intended to be used by other projects.
> Those other projects should have confidence that Marl is well tested, and not need to test it themselves.
> If they really want to build the tests, they can do so by setting MARL_BUILD_TESTS to 1.

Basically, I do not feel like it is the responsibility, nor is it desirable for the downstream projects to test marl, and to have marl included as a library is its main intended use - not to be tested by everything.

Reply:
Sounds good to me! I can accept that answer. That being said if they dont pull gtest, it wont be built anyway. I guess if `git submodule update --init --recursive` it will pull it by default, so this is sensible.